include/cosim/algorithm/simulator.hpp

```diff
@@ -133,6 +133,55 @@ class simulator : public manipulable
     virtual step_result do_step(
         time_point currentT,
         duration deltaT) = 0;
+
+    /// A type used for references to saved states (see `save_state()`).
+    using state_index = int;
```

Comment:
If a step is ongoing when `save_state` is called, should the FMU complete the step before returning it's state? Should we also save a `time_point`, or possibly use `time_point` instead of `int` as index, to be able to immediately tie a `saved_state` to the `step` it was saved from?

Reply:
The FMI specification forbids calling the state saving/restoring functions when a step is ongoing (see state machine in sec. 4.2.4 of FMI v. 2.0.4), so I don't think this is an issue. We could consider checking it, but we don't support async stepping elsewhere in libcosim yet, so this can probably wait until we do.

The `slave_simulator` class does not have the current time point as part of its internal state, so it doesn't need to save it or associate it with the lower-level FMU state either. The FMU might do so itself (though we wouldn't know about it). I'm quite certain that this is something we'll have to do when we get to saving the *full co-simulation state* in `fixed_step_algorithm`, though.